### ISSUE TITLE
Initial products cached forever

### ISSUE THREAD

**Issue by brunobasto** [DivanteLtd/vue-storefront#1460]
First of all: Congrats! This is a great project. I find the architecture really interesting and seems to be scalable. The offline implementation looks nice.

I'm having a small problem and I was hoping to get some help:

I setup everything according to the docs. I have my store up and running. I even managed to import initial data from Magento. However, after that I can't get new products to show up. None of the following commands seem to get products to show up:

`node --harmony cli.js products --partitions=1`
`node --harmony cli.js productsdelta --partitions=1`

Looking at the logs from the above commands I see that it managed to import the newer ones. My guess is that it's a cache on the storefront-api or other layer (Redis, service workers...)

Do you have any ideas on what kind of cache I should clean or where to start debugging?

Thanks in advance

**Comment by pkarw** [comment]
Hi @brunobasto! thanks for the message.

I see two ways to slove the problem:
- go to vue-storefront-api folder and run: `npm run db new` - this command will create new, empty database; then do the import and You're there
- second solution is to run the importer with additional argument; sth like:
```

node --harmony cli.js categories --partitions=1 --removeNonExistient=true
node --harmony cli.js productcategories --partitions=1
node --harmony cli.js attributes --partitions=1 --removeNonExistient=true
node --harmony cli.js taxrule --partitions=1 --removeNonExistient=true
node --harmony cli.js products --partitions=1 --removeNonExistient=true
```

The `removeNonExistient` will remove all the records from the database that were NOT PRESENT in current import.

**Comment by brunobasto** [comment]
Hey @pkarw, thanks for getting back! However, I think I didn't described my issue very well. Let me try again.

I've successfully used `removeNonExistient` to remove the sample data and replace with my own Magento data. However,  the problem is not that old products don't get deleted. The problem is that the new ones never show up. I want to sync Magento products continuously (via a cron job for example). But when I add a new product, run the cli to import the data and refresh the page, I don't see my new products on the website even though the cli told me there were new products imported.

**Comment by pkarw** [comment]
Ok! You need to run the productcategories import before products import to populate redis cache with the category assignments 

Or

Please pull the changes from github I’ve fixed this yesterday and now should be fine :)

**Comment by brunobasto** [comment]
Awesome! Will try pulling changes from Github and see how it goes. Thx

**Comment by brunobasto** [comment]
Great! It seems to work now. Thank you. Keep up the good work. Will try to contribute as soon as I get my head around the project.
